lib/utils.pm

```diff
@@ -534,9 +534,10 @@ Example:
 sub assert_screen_with_soft_timeout {
     my ($mustmatch, %args) = @_;
     # as in assert_screen
-    $args{timeout}             //= 30;
-    $args{soft_timeout}        //= 0;
-    $args{soft_failure_reason} //= $args{bugref} . ': needle(s) not found within ' . $args{soft_timeout};
+    $args{timeout}      //= 30;
+    $args{soft_timeout} //= 0;
+    my $needle_info = ref($mustmatch) eq "ARRAY" ? join(',', @$mustmatch) : $mustmatch;
+    $args{soft_failure_reason} //= $args{bugref} . ': needle(s) ' . $needle_info . ' not found within ' . $args{soft_timeout};
```

Comment:
Looks like code contained an error and it should be `$args{timeout}` here, instead of `$args{soft_failure_reason}`.

Reply:
$args{soft_failure_reason} is used in the soft-failure, I think is ok? could you rephrase it, maybe I'm missing something.

Reply:
You are right, I got confused as timeout should be bigger than soft_timeout. Sorry =)